design/operator_design.md

```diff
@@ -0,0 +1,129 @@
+# YugabyteDB Operator Design
+
+A cluster of YugaByte DB should be created using the `ybcluster.yugabytedb.yugabyte.com` custom resource definition. Below is a sample custom resource spec for creating a 3-master, 3 t-servers cluster using the CRD. The sample is followed by an explanation of different configuration options available on the YugaByte DB CRD.
+
+
+```yaml
+apiVersion: yugabytedb.yugabyte.com/v1alpha1
+kind: YBCluster
+metadata:
+  name: yugabytedb
+  namespace: yugabytedb
+spec:
+  image:
+    repository: yugabytedb/yugabyte     # Default.
+    tag: master                         # Default.
+    pullPolicy: IfNotPresent            # Default
+  tls:
+    enabled: false
+    # rootCA:
+      # cert:                           # Uncomment if tls.enabled is set to true. Specify the certificate string downloaded from Yugabyte private Enterprise Edition registry.
+      # key:                            # Uncomment if tls.enabled is set to true. Specify the certificate key string downloaded from Yugabyte private Enterprise Edition registry.
+  multiAZ:
+    enabled: false
+  master:
+    # Replica count for Master.
+    replicas: 3
+    # Mentioning network ports is optional. If some or all ports are not specified, then they will be defaulted to below-mentioned values.
+    masterUIPort: 7000                  # Optional. default value "7000".
+    masterRPCPort: 7100                 # Optional. default value "7100".
+    enableLoadBalancer: false           # Optional. default value "false". Change it to true to be able to access YugabyteDB Master UI over the internet.
+    podManagementPolicy: Parallel       # Optional. default value "Parallel", out of valid values "Parallel" and "OrderedReady". If "OrderedReady" value is specified, cluster will take more time coming up.
+    storage:
+      count: 2
```

Comment:
The `yb-master` does not need 2 disks, 1 should suffice here. cc @ramkumarvs @Arnav15

Reply:
Should we default the storage disk count to 1 for master & remove the attribute from CRD?

Reply:
@rkarthik007 Please let me know on above follow-up question & also on the few others in the review.

Reply:
Let's leave it configurable but default to 1.

Reply:
Done.